torchao/testing/model_architectures.py

```diff
@@ -179,7 +237,7 @@ def create_model_and_input_data(
         m, k, n (int): dimensions of the model and input data
     """
     if model_type == "linear":
-        model = ToyLinearModel(k, n, high_precision_dtype).to(device)
+        model = ToySingleLinearModel(k, n, device=device, dtype=high_precision_dtype)
         input_data = torch.randn(m, k, device=device, dtype=high_precision_dtype)
```

Comment:
can this be changed to use example_inputs from `model` now

e.g. `input_data = model.example_inputs(batch_size=m)[0]`